### PR TITLE
[BugFix] Update error msg when apply analytic function on array type

### LINKED ISSUE #6617: [sqlancer] improve the error message when using json type in sql having clause
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS h19;
CREATE DATABASE h19;
USE h19;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATETIME NOT NULL ,c_0_1 BOOLEAN NOT NULL ,c_0_2 CHAR(1) NOT NULL ,c_0_3 VARCHAR(1) NOT NULL ,c_0_4 VARCHAR(31) NOT NULL ,c_0_5 DATETIME NULL ,c_0_6 BOOLEAN NULL ,c_0_7 BOOLEAN NOT NULL ,c_0_8 ARRAY<INT> NULL ,c_0_9 JSON NOT NULL ,c_0_10 FLOAT NOT NULL ,c_0_11 INT NOT NULL ,c_0_12 VARCHAR(1) NULL ,c_0_13 BOOLEAN NOT NULL ,c_0_14 DATETIME NOT NULL ,c_0_15 ARRAY<STRING> NULL ,c_0_16 ARRAY<INT> NULL ,c_0_17 DOUBLE NULL , INDEX index1_c_0_0(`c_0_0`) USING BITMAP) DUPLICATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 125 day)) DISTRIBUTED BY HASH (c_0_4,c_0_2) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 CHAR(41) NULL ,c_1_1 DATE NOT NULL ,c_1_2 BOOLEAN NOT NULL ,c_1_3 BOOLEAN NULL ,c_1_4 DECIMAL(15, 7)  NULL ,c_1_5 INT NULL ,c_1_6 DECIMAL(10, 0)  NOT NULL ,c_1_7 DECIMAL(32, 5)  NOT NULL ,c_1_8 BOOLEAN NOT NULL ,c_1_9 JSON NOT NULL ,c_1_10 BOOLEAN NOT NULL ,c_1_11 LARGEINT NOT NULL ,c_1_12 DATE NOT NULL ) DUPLICATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8) PARTITION BY RANGE(c_1_1) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 337 day)) DISTRIBUTED BY HASH (c_1_5,c_1_4) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_1_11") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 BOOLEAN NULL ,c_2_1 DECIMAL(26, 3)  NOT NULL ,c_2_2 CHAR(21) NULL ,c_2_3 DECIMAL(6, 5)  NOT NULL ,c_2_4 ARRAY<INT> NOT NULL ) UNIQUE KEY (c_2_0,c_2_1) DISTRIBUTED BY HASH (c_2_0,c_2_1) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t1 VALUES ('V', '2010-01-16', false, NULL, 7.4816827, 18, 3947734674, 5946432758556727926624.31902, false, parse_json('{"JjI": "zudcW", "h": null, "VRXEdQJ": null, "XCXV8WCB": null, "jW": null, "m": null, "Bp": false, "N": null, "Q": "OFcQ", "49Z0E": true, "T": 0.5164146631319717, "5O": 0.06565913402256363, "cTDUceCOC": null, "pm": -1457301273, "zwCm": true}'), false, -651583111, '2009-12-31');
INSERT INTO t1 VALUES (NULL, '2010-01-24', false, false, 1.9543404, 21, 5294790646, 8918454128171474277131.38085, false, parse_json('{"Ij": 1408024020, "2w": {"4WR0AaE0": null, "7he": 0.37890814443960463, "6pexzE9O": 1983810596, "uHQC0JA": null, "9": -1457301273, "N": "NZ"}, "8": {}, "I": [0.4759940635256147, null, "oBGOR3037", null, null, null, null, true, null, true], "Gq": 302183908, "4Zht": {"r5": true, "Di": 0.4759940635256147, "e": null, "eZ": 0.06565913402256363, "8": 9223372036854775807, "Nxs": 0.7319920477748649, "K": "uc", "kf": false, "RysV": "3r"}, "Wb8cr": [null, 302183908, "", null, 0.10762387756264258, null, "Y27", false, null, null, true, null, "Lf8lNjiFo", "ewb", null], "dp": false, "B3a3": [false, null]}'), true, -493614498, '2010-01-30');


mysql> WITH with_t_0 as (SELECT t1_57.c_1_0, t1_57.c_1_1, t1_57.c_1_2, t1_57.c_1_3, t1_57.c_1_4, t1_57.c_1_5, t1_57.c_1_6, t1_57.c_1_7, t1_57.c_1_8, t1_57.c_1_10, t1_57.c_1_11, t1_57.c_1_12 FROM t1 AS t1_57 WHERE false) , with_t_1 as (SELECT t2_58.c_2_0, t2_58.c_2_1, t2_58.c_2_2, t2_58.c_2_3 FROM t2 AS t2_58) SELECT t1_57.c_1_0 FROM t1 AS t1_57, t2 AS t2_58, with_t_0 AS with_t_0, with_t_1 AS with_t_1 GROUP BY t2_58.c_2_0 HAVING  ( t1_57.c_1_9 ) != ( 9 )  ORDER BY t1_57.c_1_7 ASC, t1_57.c_1_8 DESC;
ERROR 1064 (HY000): Invalid agg function plan: any_value


```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE d12a28a |
+----------------------+

### LINKED ISSUE #6863: error message - Function 'last_value(`t1_86`.`c_1_2`)' not supported with OVER clause
--java.lang.AssertionError: WITH with_t_0 as (SELECT t1_86.c_1_0, t1_86.c_1_1 FROM t1 AS t1_86) , with_t_1 as (SELECT t1_86.c_1_0, t1_86.c_1_1 FROM t1 AS t1_86) SELECT t1_86.c_1_0, LAST_VALUE(t1_86.c_1_2) OVER ( PARTITION BY t1_86.c_1_2)  FROM t1 AS t1_86, with_t_0 AS with_t_0, with_t_1 AS with_t_1 WHERE false GROUP BY t1_86.c_1_2;
--	at sqlancer.common.query.QueryAdapter.checkException(QueryAdapter.java:98)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:128)
--	at sqlancer.ComparatorHelper.getResultSetFirstColumnAsString(ComparatorHelper.java:54)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPHavingOracle.check(StarRocksTLPHavingOracle.java:38)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:322)
--	at sqlancer.Main$2.run(Main.java:511)
--	at sqlancer.Main$2.runThread(Main.java:489)
--	at sqlancer.Main$2.run(Main.java:479)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
--Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Function 'last_value(`t1_86`.`c_1_2`)' not supported with OVER clause.
--	at sun.reflect.GeneratedConstructorAccessor5.newInstance(Unknown Source)
--	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
--	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
--	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
--	at com.mysql.jdbc.Util.getInstance(Util.java:408)
--	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:943)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
--	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
--	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
--	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2680)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2497)
--	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2455)
--	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1369)
--	at sqlancer.common.query.QueryAdapter.executeAndGet(QueryAdapter.java:118)

### Steps to reproduce the behavior (Required)
CREATE DATABASE h8;
USE h8;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATETIME NOT NULL ,c_0_1 DATETIME NOT NULL ,c_0_2 DATE NULL ,c_0_3 DATE NOT NULL ,c_0_4 SMALLINT NULL ,c_0_5 BOOLEAN NULL ,c_0_6 BOOLEAN NOT NULL ,c_0_7 BOOLEAN NOT NULL ,c_0_8 BOOLEAN NULL ,c_0_9 ARRAY<STRING> NULL ,c_0_10 DATETIME NOT NULL ,c_0_11 ARRAY<STRING> NOT NULL ,c_0_12 FLOAT NULL ,c_0_13 ARRAY<INT> NOT NULL ,c_0_14 BOOLEAN NOT NULL ,c_0_15 INT NOT NULL ,c_0_16 DATE NOT NULL , INDEX index1_c_0_6(`c_0_6`) USING BITMAP) UNIQUE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7,c_0_8) PARTITION BY RANGE(c_0_2) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 219 day)) DISTRIBUTED BY HASH (c_0_5,c_0_2) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_0_4") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 BOOLEAN NULL ,c_1_1 VARCHAR(1) NULL ,c_1_2 ARRAY<INT> NOT NULL ) UNIQUE KEY (c_1_0) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 VARCHAR(51) NOT NULL ,c_2_1 DOUBLE MAX NULL ,c_2_2 VARCHAR(21) MAX NOT NULL ,c_2_3 FLOAT REPLACE_IF_NOT_NULL NULL ,c_2_4 BOOLEAN REPLACE NOT NULL ) AGGREGATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE MATERIALIZED VIEW v0 AS SELECT t2_78.c_2_0, t2_78.c_2_3, t2_78.c_2_4, MAX(t2_78.c_2_1) , MAX(t2_78.c_2_2)  FROM t2 AS t2_78 GROUP BY t2_78.c_2_0, t2_78.c_2_3, t2_78.c_2_4 ORDER BY t2_78.c_2_0, t2_78.c_2_3, t2_78.c_2_4 DESC;
CREATE MATERIALIZED VIEW v0 AS SELECT t1_80.c_1_1, MIN(t1_80.c_1_0)  FROM t1 AS t1_80 GROUP BY t1_80.c_1_1 ORDER BY t1_80.c_1_1 ASC;
INSERT INTO t0 VALUES ("2010-01-03 08:00:14", "2010-01-07 08:00:14", "2010-01-23", "2010-01-23", 3, false, false, true, true, ["", "bzS", "GDkb2MWdZ", "b", "cd", "", "zQwgi", "iOYkX6tz", "Ur0", "", "K2zZdIoS", "mJhe542TI"], "2010-01-17 08:00:14", ["", ""], 0.3733675486428276, [412985474, -99331877, -1141428404, 412985474, -99331877, -1141428404, 412985474, 1529649165, 830533870, 1529649165, -17167736], false, 12, "2010-01-06"), ("2010-01-23 08:00:14", "2010-01-16 08:00:14", "2010-01-23", "2010-01-19", NULL, false, true, false, false, ["frXOu400M", "2EsS", "u4", "L", "X", "zW", "L"], "2010-01-27 08:00:14", ["", "dgfKI", "f0RMp5bj"], 0.1848353313517348, [-2081342657, 1518150261, 1228979431, -994881310, -282219595], false, 14, "2010-01-05");
INSERT INTO t2 VALUES ("ww", 0.3974359266304929, "l", 0.18119365268821286, false), ("", 0.22603276267429873, "ItRqdiMy", 8.3053387E8, false), ("", 0.18440813391504096, "D", 0.22603276267429873, true);
INSERT INTO t2 VALUES ("F9", 0.2055758386797617, "pn", 0.6448639098327671, false), ("UcBYW", 0.2055758386797617, "", 0.18119365268821286, false);

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6863
Fixes #6617

## Problem Summary(Required) ：

Prevent adding window/aggregate built-in functions of array type

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java [hunk 1]
```diff
@@ -815,13 +815,7 @@ private void initAggregateBuiltins() {
...
 
         for (Type t : Type.getSupportedTypes()) {
-            if (t.isNull()) {
-                continue; // NULL is handled through type promotion.
-            }
-            if (t.isChar()) {
-                continue; // promoted to STRING
-            }
-            if (t.isTime()) {
+            if (t.isNull() || t.isChar() || t.isTime() || t.isArrayType() || t.isJsonType() || t.isPseudoType()) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:818]: Why not replace it with `isNotNumeric`?

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:818]: Yes, it's better

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:818]: We can apply window function on `DATETIME` which is not numerical, although it does stored in numerical format

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java [hunk 2]
```diff
@@ -186,4 +188,21 @@ public void testArrayClone() throws Exception {
...
                 "  |  <slot 9> : array_contains(ARRAY<bigint(20)>[7: expr], 2)"));
     }
+
+    @Test
+    public void testArrayWindowFunction() {
+        for (String fnName : Sets.newHashSet(AnalyticExpr.LASTVALUE, AnalyticExpr.FIRSTVALUE)) {
+            try {
+                String sql = String.format("select %s(v3) over() from tarray", fnName.toLowerCase());
+                getFragmentPlan(sql);
+            } catch (Exception e) {
```
Comment (kangkaisen) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java:198]: please use ExpectedException.

you could refer to ColocateTableTest

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java [hunk 3]
```diff
@@ -815,13 +815,9 @@ private void initAggregateBuiltins() {
...
         for (Type t : Type.getSupportedTypes()) {
-            if (t.isNull()) {
-                continue; // NULL is handled through type promotion.
-            }
-            if (t.isChar()) {
-                continue; // promoted to STRING
-            }
-            if (t.isTime()) {
+            // null/char/time is handled through type promotion
+            // array/json/pseudo is not supported yet
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:819]: please add TODO, we could support array and json type later.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java (hunk 3 above)
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java:819]: ok

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java (hunk 2 above)
Comment (liuyehcf) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/ArrayTypeTest.java:198]: Indeed, but here is the thing that I want to test multiply functions in one test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|1|1|100.00%|

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**